### PR TITLE
message timestamps update onResume()

### LINKED ISSUE #2519: Time stamps very slow to update
Hi all! I'm running text secure 2.5.3 on a Moto G (XT1033 falcon_asia_ds). Running android 5.0 (LXB22,46-28). I'm using the app primarily as a texting replacement as none of my contacts use Text Secure. [Debug log](https://gist.github.com/anonymous/faa0bc4a16b4d923ca56)

I've noticed that time stamps are extremely slow in updating. If I've sent a text, then navigated out of the app, then go back into the app after a number of minutes, chances are the time stamp shows "now" or "1 min" or whatever. The time stamp only updates after maybe 10-20 seconds. I've never noticed this problem on any other texting app.

### PR DESCRIPTION
**Pull request by rhodey**
Modified conversation view and conversation list view to update message timestamps onResume.

The basics of this is that `LoaderManager.initLoader()` is now called `onResume()` instead of `onCreate()` and `LoaderManager.destroyLoader()` is now called `onPause()` so that a new loader will be created `onResume()` which will force the timestamps to update.

The null check and `-1` check on `this.recipients` and `this.threadId` is to handle the case of a new conversation thread where the list of messages would be empty-- in this case no `ListAdapter` is set so these checks prevent an NPE within two of the `LoaderManager` callbacks.

Fixes #2519

### CODE REVIEW

**Comment by rhodey** [comment]
In working on this fix I think I also discovered an extra call to `ConversationListFragment.initializeListAdapter()`

https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/ConversationListFragment.java#L146

`ConversationListFragment.setMasterSecret()` is only called by `ConversationListActivity.initializeResources()` which is called inside of the activity's `onCreate()` method.

https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/ConversationListActivity.java#L256

So `ConversationListFragment.initializeListAdapter()` is called in `onCreate()` of both the activity and the `onActivityCreated()` of the fragment-- we should be able to drop `initializeListAdapter()` from `ConversationListFragment.setMasterSecret()`

**Comment by rhodey** [comment]
dropped extra call to `initializeListAdapter()` within `ConversationListFragment.setMasterSecret()`, tested, and rebased.

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -159,6 +170,7 @@ public void reload(Recipients recipients, long threadId) {
     this.threadId   = threadId;
 
     initializeListAdapter();
+    getLoaderManager().restartLoader(0, null, this);
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: why is this necessary to do on every message sent?

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 2]
```diff
@@ -91,6 +91,18 @@ public void onNewIntent() {
...
+  @Override
+  public void onResume() {
+    super.onResume();
+    getLoaderManager().initLoader(0, null, this);
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    getLoaderManager().destroyLoader(0);
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Wouldn't we only have to destroyLoader when the threadId changes, so on a new intent or when we send our first message to a user? If not, we should just call restartLoader in onResume instead of both initLoader and destroyLoader.

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 3]
```diff
@@ -268,12 +280,16 @@ public void onFailure(Throwable error) {
 
   @Override
   public void onLoadFinished(Loader<Cursor> arg0, Cursor cursor) {
-    ((CursorAdapter)getListAdapter()).changeCursor(cursor);
+    if (this.recipients != null && this.threadId != -1) {
```
Comment (rhodey) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: These checks are to prevent an NPE on `getListAdapter()` which is not set in the case that this fragment is for a new message (identifiable by null recipients and threadId of -1). Alternatively could just check that `getListAdapter() != null`, not sure which is better style / more understandable.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 2 above)
Comment (rhodey) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: `initLoader()` should be called before any other `LoaderManager` method because it handles some needed initialization but in addition to this it also creates a new loader. So if we placed `restartLoader()` in `onResume()` we'd have to place `initLoader()` in either `onCreate()` or `onResume()` and this would result in one loader being needlessly created.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 1 above)
Comment (rhodey) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: thanks, this should be changed to restart the loader only if the `threadId` changes which handles the case where the fragment represented a new conversation and because of this didn't create a loader.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 2 above)
Comment (rhodey) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Also, if we agree that `initLoader()` should happen in `onResume()` to avoid wasting a loader then `destroyLoader()` must happen in `onPause()` as above so that the next call to `initLoader()` in `onResume()` will notice that it's loader is gone, create a new one, and ultimately update the timestamp.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 1 above)
Comment (rhodey) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: actually, I'll need to check this. my reasoning for this was that previous to these changes the entire list adapter and loader were reinitialized after every message send.

https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/ConversationFragment.java#L161
